### ISSUE TITLE
Add ignoreUnits: [] to number-max-precision

### ISSUE THREAD

**Issue by ZachTRice** [stylelint/stylelint#2785]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Feature request:

Add `ignore: ["%", "ch", "cm", "em", "ex", "in", "mm", "pc", "pt", "px", "rem", "vh", "vmin", "vmax", "vw" ]`

> Which rule, if any, is this issue related to?

number-max-precision

> What CSS is needed to reproduce this issue?

```css
// New browsers are not effected by sub-pixel rounding; thus % rules should be ignored.
.foo {
  width: 66.666667%;
}

// If not an ignored unit (i.e. %), ensure it is equal to or less than the specified precision (in this case 1)
.bar {
  font-size: 9.5px;
}
```

> What stylelint configuration is needed to reproduce this issue?

```
"number-max-precision": [1, {
  ignore: "%"
}],
```

> Which version of stylelint are you using?

8.0.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

grunt-postcss stylelint plugin; also tested with grunt-stylelint

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

I would like to constrain all units except for % based units to 1 number-max-precision. What I am suggesting is adding an option to ignore specific units, such as pixels, percentages, etc so that they are not constrained. 

> What actually happened (e.g. what warnings or errors you are getting)?

No errors. All units were constrained to 1 number-max-precision as I set but no additional functionality was available to ignore specific units.

**Comment by jeddy3** [comment]
@ZachTRice Thanks for the request and for using the template.  Sorry about the delay triaging this issue (holidays and what not...).

Also, thanks for illustrating your use case so clearly. I agree that the addition of a secondary option is what you need. As [per the guidelines](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#keyword-ignore-and-except), it'd be called `ignoreUnits: []` rather than `ignore: []`.

I'm going to label this issue as "new option" and "help wanted" (and "beginner friendly" as the rule makes use of the `postcss-value-parser` which should make checking the unit pretty easy). Please consider contributing this option if you have time. There's [a section in the Developer Guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-an-option-to-an-existing-rule) on how to get started.

**Comment by GorwinJororis** [comment]
@jeddy3 I believe this issue is no longer matter.

**Comment by jeddy3** [comment]
@GorwinJororis Thanks!
